app/models/course_data/assignment.rb

```diff
@@ -105,6 +108,19 @@ def peer_review_pagename
     "#{sandbox_pagename}/#{user.username}_Peer_Review"
   end
 
+  def update_sandbox_url(new_url)
+    return unless new_url
+    return unless user
+    # creates and initializes instance variable (edit_sandbox_url)
+    # used to bypass the before_save (set_sandbox_url) method.
+    @edit_sandbox_url = true
```

Comment:
This should not be necessary; as long as the `sandbox_url` is already present, `set_sandbox_url` will not change it because of the `return if sandbox_url.present?` line.